typegen/src/typegen/validation.rs

```diff
@@ -53,9 +57,10 @@ pub fn validate_substitutes_and_derives_against_registry(
 
     for (path, sub) in substitutes.iter() {
         if !registry_contains_type_path(types, path) {
-            error
-                .substitutes_for_unknown_types
-                .push((path.clone(), sub.path().clone()))
+            error.substitutes_for_unknown_types.push((
+                path_segments_to_syn_path(path).expect("Path in Substitutes should not be empty."),
```

Comment:
Just to check; do we ensure that the path in substittues is not empty when they are provided?

Reply:
Yes, they should never be empty because they come from the `TypeSubstitutes` struct that the user passes in. And the only legal way to create `TypeSubstitutes`  is by inserting types into it, which errors in case the type has an empty path. (every code path calls parse_path_param_mapping() which errors if the path is empty).

Reply:
Cool, thanks!